assets/source/setup-guide/app/constants.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * Enum of general label status.
+ *
+ * @readonly
+ * @enum {string}
+ */
+export const LABEL_STATUS = Object.freeze( {
+	PENDING: 'pending',
+	SUCCESS: 'success',
+} );
+
+/**
+ * Enum of general process status.
+ *
+ * @readonly
+ * @enum {string}
+ */
+export const PROCESS_STATUS = Object.freeze( {
+	...LABEL_STATUS,
+	IDLE: 'idle',
+	ERROR: 'error',
+} );
```

Comment:
💅  I feel like we shouldn't have `LABEL_STATUS`, it is used in `if` condition check in `VerifyButton` which feels not very straightforward and readable.

I would prefer we have `PROCESS_STATUS` only like this: 

```suggestion
/**
 * Enum of general process status.
 *
 * @readonly
 * @enum {string}
 */
export const PROCESS_STATUS = Object.freeze( {
	IDLE: 'idle',
	PENDING: 'pending',
	SUCCESS: 'success',
	ERROR: 'error',
} );
```

Note that the line order also indicate the sequence: it starts at idle, and then pending, and then success or error, which IMO is easy to understand and less mental overhead.

Reply:
> I feel like we shouldn't have `LABEL_STATUS`, it is used in `if` condition check in `VerifyButton` which feels not very straightforward and readable.

`LABEL_STATUS` is also used to indicate the `status` prop that it accepts in the `<StatusLabel>` component and its JSDoc. And `PROCESS_STATUS` is not the status set accepted by this component.

https://github.com/woocommerce/pinterest-for-woocommerce/blob/70205b028a0c460753f32ea58057ef5e3dbe1b75/assets/source/setup-guide/app/components/StatusLabel/index.js#L30-L37

I tried back and forth to get `@enum` to be more flexible with exporting/importing references, but it doesn't seem to work. For example, the following way would make the type of `status` appear as `any | any`

```js
/**
 * @typedef {import('../../constants').PROCESS_STATUS} PROCESS_STATUS
 */
/**
 * @param {Object} props React props.
 * @param {PROCESS_STATUS.PENDING|PROCESS_STATUS.SUCCESS} props.status Label status.
 */
```

![image](https://user-images.githubusercontent.com/17420811/144964966-8482da2c-5286-4d66-922f-81fd499e5026.png)

---

> Note that the line order also indicate the sequence: it starts at idle, and then pending, and then success or error, which IMO is easy to understand and less mental overhead.

They have finite-state machine relation, in which the order of states does not always switch according to the above arrangement. I don't think that the finite-state machine relation should be implicitly stated by the order of appearance of the code to indicate the order of the states. Because it may be only one (happy path) of the state transition paths, but not all of them.

Reply:
> > I feel like we shouldn't have `LABEL_STATUS`, it is used in `if` condition check in `VerifyButton` which feels not very straightforward and readable.
> 
> `LABEL_STATUS` is also used to indicate the `status` prop that it accepts in the `<StatusLabel>` component and its JSDoc.
> 

Sorry, I should have been clearer 😅 My comment here was related to https://github.com/woocommerce/pinterest-for-woocommerce/pull/286#discussion_r762906837 and https://github.com/woocommerce/pinterest-for-woocommerce/pull/286#discussion_r763011970 (changing `StatusLabel` into `IconLabel`). With those changes, we wouldn't need `LABEL_STATUS`.